services/src/test/scala/cromwell/pipeline/service/GitLabProjectVersioningTest.scala

```diff
@@ -45,40 +61,60 @@ class GitLabProjectVersioningTest extends AsyncWordSpec with ScalaFutures with M
 
       "return new active Project with 201 response" taggedAs Service in {
         val project = withRepoProject
-        when(request(activeProject)).thenReturn(Future.successful(Response(HttpStatusCodes.Created, EmptyBody, Map())))
+        when(request(activeProject)).thenReturn(
+          Future.successful(
+            Response[Project](HttpStatusCodes.Created, SuccessResponseBody[Project](withRepoProject), Map())
+          )
+        )
+
         gitLabProjectVersioning.createRepository(activeProject).map {
           _ shouldBe Right(project)
         }
       }
 
       "throw new VersioningException with 400 response" taggedAs Service in {
-        when(request(activeProject))
-          .thenReturn(Future.successful(Response(HttpStatusCodes.BadRequest, EmptyBody, Map())))
+        when(request(activeProject)).thenReturn(
+          Future.successful(Response[Project](HttpStatusCodes.BadRequest, FailureResponseBody(EmptyBody), Map()))
+        )
         gitLabProjectVersioning.createRepository(activeProject).map {
           _ shouldBe Left(VersioningException("The repository was not created. Response status: 400"))
         }
       }
     }
 
     "getProjectVersions" should {
-      val dummyVersionsJson: String = s"[${Json.stringify(Json.toJson(dummyVersion))}]"
 
       def request(project: Project) =
-        mockHttpClient.get(
-          url = gitLabConfig.url + "projects/" + project.repository.get.value + "/repository/tags",
-          headers = gitLabConfig.token
-        )
+        mockHttpClient.get[Seq[Version]](
+          url = MockitoMatchers.eq(gitLabConfig.url + "projects/" + project.repository.get.value + "/repository/tags"),
+          headers = MockitoMatchers.any[Map[String, String]],
+          params = MockitoMatchers.any[Map[String, String]]
+        )(MockitoMatchers.any[ExecutionContext], MockitoMatchers.any[Reads[Seq[Version]]])
 
       "return list of Project versions with 200 response" taggedAs Service in {
-        when(request(withRepoProject))
-          .thenReturn(Future.successful(Response(HttpStatusCodes.OK, dummyVersionsJson, Map())))
+        when(request(withRepoProject)).thenReturn(
+          Future.successful(
+            Response[Seq[Version]](HttpStatusCodes.OK, SuccessResponseBody[Seq[Version]](Seq(dummyVersion)), Map())
+          )
+        )
+
+        println("gitLabProjectVersioning=" + gitLabProjectVersioning)
+        println("withRepoProject=" + withRepoProject)
+        println(
+          "getProjectVersions(withRepoProject)=" + gitLabProjectVersioning.getProjectVersions(
+            withRepoProject
+          )
+        )
+        println()
```

Comment:
Why are there so many `println`s?

Reply:
deleted, I will be carefull after resolving conflicts =(